modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/describer/MuleExtensionAnnotationParser.java

```diff
@@ -213,23 +210,47 @@ private static boolean shouldAdvertise(MetadataType parameterType, Map<Class<? e
         return map;
     }
 
-    public static void parseDisplayAnnotations(AnnotatedElement annotatedElement, HasModelProperties parameter)
+    private static void parseDisplayAnnotations(AnnotatedElement annotatedElement, DisplayModelPropertyBuilder builder)
     {
         Password passwordAnnotation = annotatedElement.getAnnotation(Password.class);
         if (passwordAnnotation != null)
         {
-            parameter.withModelProperty(PasswordModelProperty.KEY, new ImmutablePasswordModelProperty());
+            builder.withPassword(true);
         }
         Text textAnnotation = annotatedElement.getAnnotation(Text.class);
         if (textAnnotation != null)
         {
-            parameter.withModelProperty(TextModelProperty.KEY, new ImmutableTextModelProperty());
+            builder.withText(true);
         }
+    }
+
+    public static void parsePlacementAnnotation(AnnotatedElement annotatedElement, DisplayModelPropertyBuilder builder)
+    {
         Placement placementAnnotation = annotatedElement.getAnnotation(Placement.class);
         if (placementAnnotation != null)
         {
-            parameter.withModelProperty(PlacementModelProperty.KEY, new ImmutablePlacementModelProperty(placementAnnotation.order(), placementAnnotation.group(), placementAnnotation.tab()));
+            builder.order(placementAnnotation.order()).groupName(placementAnnotation.group()).tabName(placementAnnotation.tab());
         }
     }
 
+    public static void parseDisplayNameAnnotation(AnnotatedElement annotatedElement, String fieldName, DisplayModelPropertyBuilder builder)
+    {
+        DisplayName displayNameAnnotation = annotatedElement.getAnnotation(DisplayName.class);
+        String displayName = (displayNameAnnotation != null) ? displayNameAnnotation.value() : StringUtils.join(StringUtils.splitByCharacterTypeCamelCase(StringUtils.capitalize(fieldName)), ' ');
+        builder.displayName(displayName);
+    }
+
+    public static void parseDisplayAnnotations(AnnotatedElement annotatedElement, String name, ParameterDescriptor parameter)
+    {
+        parseDisplayAnnotations(annotatedElement, name, parameter, DisplayModelPropertyBuilder.create());
+    }
+
+    public static void parseDisplayAnnotations(AnnotatedElement annotatedElement, String name, ParameterDescriptor parameter, DisplayModelPropertyBuilder builder)
```

Comment:
shouldn't this be the only public method of the familiy and the rest just be private helpers?